### PR TITLE
fix: user agents in webviews

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-1081] 

### Description

A while back we fixed a bug in our android webviews by changing the way we pass user agent: 
https://github.com/artsy/eigen/pull/9354 
This was due to a bug in the react-native-webview dependency:
https://github.com/react-native-webview/react-native-webview/issues/2949

This unfortunately also changed the format of our user agent to append our custom user agent to the browser user agent:
e.g: `Mozilla/5.0 (Linux; Android 14; Pixel 7 Pro Build/AP2A.240705.004; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/126.0.6478.134 Mobile Safari/537.36 Mozilla/5.0 (Linux; Android 14; Pixel 7 Pro Build/AP2A.240705.004; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/126.0.6478.134 Mobile Safari/537.36 Artsy-Mobile/8.44.0 Eigen/1674645751/8.44.0`  vs just `Artsy-Mobile/8.44.0 Eigen/1674645751/8.44.0` 

And this broke analytics reporting for webview actions like checkout on Android (possibly impacted iOS to a lesser extent? 🤔 ) 

This pr applies workaround recommended here: 
https://github.com/react-native-webview/react-native-webview/pull/3133

which will hopefully get our user agent back to the old format and keep the back button bug fixed.

### Before

<details>

<summary>
iOS user agent
</summary>

ios-user-agent-before-short

</details>

<details>

<summary>
Android user agent
</summary>

android-user-agent-before

</details>

### After 

<details>
<summary>
iOS user agent
</summary>

ios-user-agent-after

</details>

<details>
<summary>
Android user agent
</summary>

android-user-agent-after

</details>

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- fix user agent in webviews - brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-1081]: https://artsyproduct.atlassian.net/browse/PHIRE-1081?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (fix user agent in webviews - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f2abb34200e4679713fbabd1e403291b3fabc3ac
</p>

**Review by MrSltun — APPROVED** [review]
Nice catch and great solution Brian! 🌟

File: src/app/Components/ArtsyWebView.tests.tsx [hunk 1]
```diff
@@ -150,13 +152,27 @@ describe("ArtsyWebViewPage", () => {
...
+      Platform.OS = "android"
+      const tree = render()
+      const source = webViewProps(tree).source as any
+      expect(source).toHaveProperty("headers")
+      expect(source?.headers["User-Agent"]).toBe(
+        `Artsy-Mobile android Artsy-Mobile/${appJson().version} Eigen/some-build-number/${
+          appJson().version
+        }`
+      )
+    })
```
Comment (gkartalis) [inline src/app/Components/ArtsyWebView.tests.tsx:175]: 👏

File: src/app/Components/ArtsyWebView.tsx [hunk 2]
```diff
@@ -268,9 +268,18 @@ export const ArtsyWebView = forwardRef<
...
+            // see: https://github.com/react-native-webview/react-native-webview/pull/3133
+            ...(Platform.OS === "android" && {
+              headers: {
+                "User-Agent": userAgent,
+              },
+            }),
+          }}
           style={{ flex: 1 }}
-          applicationNameForUserAgent={userAgent}
+          userAgent={Platform.OS === "ios" ? userAgent : undefined}
```
Comment (gkartalis) [inline src/app/Components/ArtsyWebView.tsx:282]: 👏

**Review by gkartalis — APPROVED** [review]

**Pull request merged**